### ISSUE TITLE
How to include non-ascci characters (ie latin1) in a file path

### ISSUE THREAD

**Issue by msgoussi** [yihui/knitr#1983]
I would like to include a picture with a file name (C\xf4te d'Ivoire_Polar)

````
---
title: "Untitled"
output: 
  pdf_document: 
    latex_engine: xelatex
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r pressure, echo=FALSE, fig.cap="A caption", out.width = '100%'}
knitr::include_graphics("Côte d'Ivoire_Polar.eps")
```
````

C:/Program Files/RStudio/bin/pandoc/pandoc" +RTS -K512m -RTS trial.utf8.md --to latex --from markdown+autolink_bare_uris+tex_math_single_backslash --output trial.tex --lua-filter "D:\R\R-4.0.5\library\rmarkdown\rmarkdown\lua\pagebreak.lua" --lua-filter "D:\R\R-4.0.5\library\rmarkdown\rmarkdown\lua\latex-div.lua" --self-contained --highlight-style tango --pdf-engine xelatex --variable graphics --variable "geometry:margin=1in" 
output file: trial.knit.md

! xdvipdfmx:fatal: Image inclusion failed for "Côte d'Ivoire_Polar.eps".

! No output PDF file written.

Error: LaTeX failed to compile trial.tex. See https://yihui.org/tinytex/r/#debugging for debugging tips. See trial.log for more info.
In addition: Warning message:
In has_crop_tools() : 
Tool(s) not installed or not in PATH: pdfcrop, ghostcript
-> As a result, figure cropping will be disabled.
Execution halted

**Comment by cderv** [comment]
🤔  I am not sure LaTeX is keen on non ascci file path. 

Is this something you need to keep ? 
General advice would be to no use special chars in filename, avoid spaces too when possible. You'll always find an edge case somewhere. 

I'll have a look as I am on a Latin1 system too, but not sure if there will be a better solution than just use more generic filename.

**Comment by cderv** [comment]
I can reproduce with `eps` file specifically. 

````markdown
---
title: "Untitled"
output: 
  pdf_document: 
    latex_engine: xelatex
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
setEPS()
postscript("Côte d'Ivoire_Polar.eps")
plot(mtcars)
dev.off()
```


```{r pressure, echo=FALSE, fig.cap="A caption", out.width = '100%'}
knitr::include_graphics("Côte d'Ivoire_Polar.eps")
```
````

I believe this is because `xdvipdfmx` tool is called and it won't handle correctly non ascii chars (maybe).
This tools is responsible for the conversion of `.eps` to `.pdf` by `xelatex`
If you use directly png or even pdf it will work ok. 

````markdown
---
title: "Untitled"
output: 
  pdf_document: 
    latex_engine: xelatex
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
pdf("Côte d'Ivoire_Polar.pdf")
plot(mtcars)
dev.off()
```


```{r pressure, echo=FALSE, fig.cap="A caption", out.width = '100%'}
knitr::include_graphics("Côte d'Ivoire_Polar.pdf")
```
````

If you set `options(tinytex.engine_args = "-no-pdf")` so that `xelatex` don't convert to pdf, it will also works. 

So basically the issue is with 

* `.eps` file is used with `xelatex`
* processed by `xdvipdfmx` to become `.pdf`
* Special chars are not handled well by this tool. 

Changing the filename or the extension will fix this issue I believe. Changing `latex_engine` will also fix this.

Is that something possible ?

**Comment by msgoussi** [comment]
Changing the filename is the best solution for me.

**Comment by cderv** [comment]
I don't know advantage or not to use `.eps` file - I have never used them before. For LaTeX, pdf is fine I guess. 
I usually use png as image format. 

Anyway, you can change the format or the filename and it should work. Did you try it ?

**Comment by msgoussi** [comment]
I have tried to rename the file and it is working perfectly. 
However, this issue did not exit in the previous tinytext, ie this happened after i update tinytext a week ago.
Thanks for your help.

**Comment by cderv** [comment]
> However, this issue did not exit in the previous tinytext, ie this happened after i update tinytext a week ago.
Thanks for your help.

It is possible the update of your TeX Live distribution with TinyTex did update some components (TeX Live 2021 has been out since 01/04/2021)

**Comment by cderv** [comment]
I'll close this issue as I don't think there is anything we can do currently in **knitr** for this. 
Thanks!
